src/ds/d500/hw_monitor_extended_buffers.cpp

```diff
@@ -7,6 +7,20 @@
 
 namespace librealsense
 {
+    std::string hw_monitor_extended_buffers::get_firmware_version_string(const std::vector<uint8_t>& buff, size_t index, size_t length) const
+    {
+        std::stringstream formattedBuffer;
+        std::string s = "";
+        std::vector<uint16_t> buff_converted(buff.data() + index, buff.data() + index + length * sizeof(uint16_t));
```

Comment:
While functionally correct, this is wasteful. Isn't there a way to read the data without incurring unnecessary allocations, and copies?